### ISSUE TITLE
[v2 FEATURE]: Allow empty test set in split sizes argument

### ISSUE THREAD

**Issue by am2145** [chemprop/chemprop#1165]
**Is your feature request related to a problem? Please describe.**
It would be nice to support passing a split to the "--split-size" argument in `train.py` that has an empty test set, for example [0.9, 0.1, 0]. Currently, this results in the following Astartes error message: 

![Image](https://github.com/user-attachments/assets/42d0731b-7878-42a3-a274-6c820d1d23ad)

Which I believe is due to the interface to Astartes in `splitting.py` supporting an empty validation set but not an empty test set. 

**Use-cases/examples of this new feature**
This is a very similar request to #1080 except with replicates instead of folds, but the motivation/idea is essentially the same. There are cases where we have our own held-out test sets that we use to compute metrics and evaluate trained models. It would be nice if Chemprop could create multiple replicate splits of just training and validation sets for model training purposes via the "--split-size" argument, without the expectation that our test set is included in the input data. I understand that this is likely achievable through the splits file option, but I think having this be accessible through "--split-size" is more convenient to the user as we don't have to create each of the splits and place them in the JSON file ourselves. 

**Desired solution/workflow**
Naively swapping the val and test indices in the example training notebook when calling `make_split_indices` seems to work as expected, and I believe the training CLI can already largely handle an empty test set. I think either modifying the `make_split_indices` function to also allow setting only training and validation sets or having a simple check and swapping the assignment of the validation and test indices if a test size of 0 is passed would achieve the desired workflow, but I have not done a robust check. 

I'd appreciate any thoughts on this or alternate suggestions. Thank you!

**Comment by JacksonBurns** [comment]
I think we can implement this as part of our eventual solution to this request: https://github.com/chemprop/chemprop/issues/1080

**Comment by KnathanM** [comment]
Closed by #1243
